gdpr/gdpr.go

```diff
@@ -24,45 +21,22 @@ type Permissions interface {
 	// Determines whether or not to send PI information to a bidder, or mask it out.
 	//
 	// If the consent string was nonsensical, the returned error will be an ErrorMalformedConsent.
-	AuctionActivitiesAllowed(ctx context.Context, bidderCoreName openrtb_ext.BidderName, bidder openrtb_ext.BidderName, PublisherID string, gdprSignal Signal, consent string, weakVendorEnforcement bool, aliasGVLIDs map[string]uint16) (allowBidReq bool, passGeo bool, passID bool, err error)
+	AuctionActivitiesAllowed(ctx context.Context, bidderCoreName openrtb_ext.BidderName, bidder openrtb_ext.BidderName, PublisherID string, gdprSignal Signal, consent string, aliasGVLIDs map[string]uint16) (allowBidReq bool, passGeo bool, passID bool, err error)
 }
 
-// Versions of the GDPR TCF technical specification.
-const (
-	tcf2SpecVersion uint8 = 2
-)
-
 // NewPermissions gets an instance of the Permissions for use elsewhere in the project.
-func NewPermissions(ctx context.Context, cfg config.GDPR, vendorIDs map[openrtb_ext.BidderName]uint16, client *http.Client) Permissions {
+func NewPermissions(cfg config.GDPR, tcf2Config TCF2ConfigReader, vendorIDs map[openrtb_ext.BidderName]uint16, fetcher VendorListFetcher) Permissions {
 	if !cfg.Enabled {
 		return &AlwaysAllow{}
 	}
 
-	gdprDefaultValue := SignalYes
-	if cfg.DefaultValue == "0" {
-		gdprDefaultValue = SignalNo
-	}
-
-	purposeConfigs := map[consentconstants.Purpose]config.TCF2Purpose{
-		1:  cfg.TCF2.Purpose1,
-		2:  cfg.TCF2.Purpose2,
-		3:  cfg.TCF2.Purpose3,
-		4:  cfg.TCF2.Purpose4,
-		5:  cfg.TCF2.Purpose5,
-		6:  cfg.TCF2.Purpose6,
-		7:  cfg.TCF2.Purpose7,
-		8:  cfg.TCF2.Purpose8,
-		9:  cfg.TCF2.Purpose9,
-		10: cfg.TCF2.Purpose10,
-	}
-
 	permissionsImpl := &permissionsImpl{
-		cfg:              cfg,
-		gdprDefaultValue: gdprDefaultValue,
-		purposeConfigs:   purposeConfigs,
-		vendorIDs:        vendorIDs,
-		fetchVendorList: map[uint8]func(ctx context.Context, id uint16) (vendorlist.VendorList, error){
```

Comment:
The vendor list fetcher was previously created when the single GDPR permissions object was created at startup in the router and was tightly coupled. Now, since we need to instantiate a permissions object at startup for cookie_sync and setuid endpoints and also on a per request basis for the ORTB2 endpoints, I had to decouple the fetcher from the permissions object as I figured it made sense to continue having just a single instance of the vendor list fetcher. As a result, the fetcher is created at startup using the same functional programming style we've always used to create it and then passed into the GDPR permissions object created at startup and those created during the auction.